### PR TITLE
[CARBONDATA-2514] Added condition to check for duplicate column names

### PR DESCRIPTION
**Pull request by kunal642**
1. Duplicate columns check was not present.
2. indexFileReader was not being closed due to which index file could not be deleted.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 
 - [x] Any backward compatibility impacted?
 
 - [x] Document update required?

 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by kunal642** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5052/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6047/

File: core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java [hunk 1]
```diff
@@ -2382,6 +2382,7 @@ static DataType thriftDataTyopeToWrapperDataType(
     CarbonIndexFileReader indexFileReader = new CarbonIndexFileReader();
     indexFileReader.openThriftReader(indexFilePath);
     org.apache.carbondata.format.IndexHeader readIndexHeader = indexFileReader.readIndexHeader();
+    indexFileReader.closeThriftReader();
```
Comment (manishgupta88) [inline core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java]: Add a try finally block and move closing of thrift reader to finally block

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6066/

**Comment by kunal642** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5067/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6068/

**Comment by manishgupta88** [comment]
LGTM

**Pull request closed without merging**